python/lsst/meas/base/applyApCorr.py

```diff
@@ -0,0 +1,156 @@
+
+class ApCorrInfo(object):
+    """!Catalog field names and keys needed to aperture correct a particular flux
+    """
+    def __init__(self, schema, algName):
+        """!Construct an ApCorrInfo and add fields to the schema
+
+        @param[in,out] schema  source catalog schema;
+            three fields are used to generate keys:
+            - {algName}_flux
+            - {algName}_fluxSigma
+            - {algName}_fluxFlag
+            three fields are added:
+            - {algName}_ApCorr
+            - {algName}_ApCorrSigma
+            - {algName}_flag_apcorr
+        @param[in] algName  name of flux measurement algorithm, e.g. "base_PsfFlux"
+        """
+        self.algName = algName  # name of flux measurement algorith
+        self.fluxName = algName + "_flux"           # name of flux field
+        self.fluxSigmaName = algName + "_fluxSigma" # name of flux sigma field
+        self.fluxKey = schema.find(self.fluxName).key           # key to flux field
+        self.fluxSigmaKey = schema.find(self.fluxSigmaName).key # key to flux sigma field
+        self.fluxFlagKey = schema.find(algName + "_flag").key   # key to flux flag field
+        self.apCorrKey = schema.addField(
+            algName + "_ApCorr",
+            doc = "aperture correction applied to %s" % (algName,),
+            type = float,
+        )
+        self.apCorrSigmaKey = schema.addField(
+            algName + "_ApCorr_Sigma",
+            doc = "aperture correction applied to %s" % (algName,),
+            type = float,
+        )
+        self.apCorrFlagKey = schema.addField(
+            algName + "_flag_apcorr",
+            doc = "set if unable to aperture correct %s" % (algName,),
+            type = "Flag",
+        )
+
+class ApplyApCorrConfig(lsst.pex.config.Config):
+    ignoreList = lsst.pex.config.ListField(
+        doc = "flux measurement algorithms in getApCorrNameSet() to ignore;" +
+            " if a name is listed that does not appear in getApCorrNameSet() then a warning is logged",
+        dtype = str,
+        optional = False,
+        default = (),
+    )
+    doFlagApCorrFailures = lsst.pex.config.Field(
+        doc = "set the general failure flag for a flux when it cannot be aperture-corrected?",
+        dtype = bool,
+        default = True,
+    )
+
+
+class ApplyApCorrTask(lsst.pipe.base.Task):
+    """!Apply aperture corrections
+    """
+    ConfigClass = ApplyApCorrConfig
+    _DefaultName = "applyApCorr"
+
+    def __init__(self, schema, **kwds):
+        lsst.pipe.base.Task.__init__(self, **kwds)
+
+        self.apCorrInfoDict = dict()
+        apCorrNameSet = getApCorrNameSet()
+        ignoreSet = set(self.config.ignoreList)
+        missingNameSet = ignoreSet - set(apCorrNameSet)
+        if missingNameSet:
+            self.log.warn("Fields in ignoreList that are not in fluxCorrectList: %s" %
+                (sorted(list(missingNameSet)),))
+        for algName in apCorrNameSet - ignoreSet:
```

Comment:
Agreed. I tried to eliminate all use of "algorithmn name", replacing it with the clumsier but more accurate "flux field name prefix", but apparently had not fully succeeded. I changed "algName" to "name" as "apCorrName" did not seem to provide useful additional information in the contexts in which it was used.